diff -ur libaria2-1.35.0/src/crypto_endian.h libaria2-1.35.0.patched/src/crypto_endian.h
--- libaria2-1.35.0/src/crypto_endian.h	2017-11-08 13:39:06.000000000 +0100
+++ libaria2-1.35.0.patched/src/crypto_endian.h	2018-03-27 16:56:55.264371501 +0200
@@ -33,9 +33,13 @@
 #  define BIG_ENDIAN 4321
 #  define BYTE_ORDER LITTLE_ENDIAN
 #else // !  defined(_WIN32) || defined(__INTEL_COMPILER) || defined (_MSC_VER)
+#if defined(__ANDROID__)
+#include <endian.h>
+#else // defined(__ANDROID__)
 #  ifdef HAVE_SYS_PARAM_H
 #    include <sys/param.h>
 #  endif // HAVE_SYS_PARAM_H
+#endif // defined(__ANDROID__)
 #endif // !  defined(_WIN32) || defined(__INTEL_COMPILER) || defined (_MSC_VER)
 
 #if !defined(LITTLE_ENDIAN) || !defined(BIG_ENDIAN) || !defined(BYTE_ORDER) || \
